File: src/learning_communication/scripts/c_talker.py
#! /usr/bin/python
import rospy
from learning_communication.msg import Person

def c_talker():
    rospy.init_node("c_talker", anonymous=True)
    puber = rospy.Publisher("c_talker", Person, queue_size=100)
    loop_rate = rospy.Rate(10)
    age = 0
    while not rospy.is_shutdown():
        msg = Person(name='Fred', age=23)
        rospy.loginfo('publish: name:{0}, age:{1}'.format(msg.name, msg.age))
        puber.publish(msg)
        age += 1
        loop_rate.sleep()


if __name__ == "__main__":
    try:
        c_talker()
    except rospy.ROSInterruptException:
        pass

File: src/learning_communication/src/c_talker.cpp
#include "ros/ros.h"
#include <sstream>
#include "learning_communication/Person.h"


int main(int argc, char *argv[])
{
    ros::init(argc, argv, "c_talker");
    ros::NodeHandle n;
    ros::Publisher puber = n.advertise<learning_communication::Person>("c_puber", 1000);
    ros::Rate loop_rate(10);

    while(ros::ok())
    {
        learning_communication::Person msg;
        msg.name = "Fred!";
        msg.age = 23;
        puber.publish(msg);
        ROS_INFO("publish msg: %s %d", msg.name.c_str(), msg.age);
        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}
